### PR TITLE
Add Fastify support

### LINKED ISSUE #2: Support Fastify
We're currently migrating from `express` to `fastify`, would love to keep using this library!

https://www.fastify.io/

### PR DESCRIPTION
**Pull request by remcohaszing**
This adds support for Fastify 3.x.

Closes #2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,6 +46,7 @@
     "eslint-plugin-prettier": "^3.1.3",
     "eslint-plugin-simple-import-sort": "^5.0.3",
     "express": "^4.17.1",
+    "fastify": "^2.14.1",
```
Comment (SimenB) [inline package.json:49]: could you verify with `fastify@3` as well?

Comment (remcohaszing) [reply package.json]: Done

File: index.ts [hunk 2]
```diff
@@ -54,28 +126,18 @@ interface KoaLike {
...
-        } else {
-          resolve();
-        }
-      });
-    });
+  inst.defaults.baseURL = String(new URL(baseURL || '', uri));
+  inst.close = async (): Promise<void> => {
+    inst.defaults.baseURL = baseURL;
+    await close();
+    await new Promise((resolve) => setTimeout(() => resolve(), 3e3));
```
Comment (SimenB) [inline index.ts:138]: why is this 3 second timeout here? It makes every test take 3s extra

Comment (remcohaszing) [reply index.ts:138]: I’m not sure how that got in. I removed it and made a patch release.

**Pull request merged**